src/components/entityset/EntitySetList.js

```diff
@@ -19,11 +25,96 @@ export class EntitySetList extends React.Component {
     });
 
     return (
-      <div className={styles.entitySetList}>
+      <div className={classnames(styles.entitySetList, this.props.className)}>
         {entitySetList}
       </div>
     );
   }
 }
 
-export default EntitySetList;
+export const FilterParamsPropType = PropTypes.shape({
+  keyword: PropTypes.string,
+  propertyTypeIds: PropTypes.arrayOf(React.PropTypes.string),
+  entityTypeId: PropTypes.string
+});
+
+export const filteredEntitySetListPropTypes = Object.assign({}, baseEntitySetListPropTypes, {
+  filterParams: FilterParamsPropType.isRequired,
+  onFilterUpdate: PropTypes.func
+});
+
+export class FilteredEntitySetList extends React.Component {
+  static propTypes = filteredEntitySetListPropTypes;
+
+  constructor(props) {
+    super(props);
+
+    this.onKeywordChange = this.onKeywordChange.bind(this);
```

Comment:
we've been using an alternative approach. instead of:

```
constructor() {
  this.onKeywordChange = this.onKeywordChange.bind(this);
}
...
onKeywordChange() {}
```

... we do:

```
onKeywordChange = () => {}
```